File: 58.length-of-last-word.js
/*
 * @lc app=leetcode id=58 lang=javascript
 *
 * [58] Length of Last Word
 */

// @lc code=start
/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLastWord = function (s) {
  let res = 0;
  for (let i = s.length - 1; i >= 0; --i) {
    if (s[i] === " ") {
      if (res > 0) {
        return res;
      }
    } else {
      res += 1;
    }
  }
  return res;
};
// @lc code=end

File: 58.length-of-last-word.go
/*
 * @lc app=leetcode id=58 lang=golang
 *
 * [58] Length of Last Word
 */

package leetcode

// @lc code=start
func lengthOfLastWord(s string) int {
	res := 0
	for i := len(s) - 1; i > -1; i-- {
		if s[i] == ' ' {
			if res > 0 {
				return res
			}
		} else {
			res++
		}
	}
	return res
}

// @lc code=end
